File: python/3264.final-array-state-after-k-multiplication-operations-i.py
import heapq


class Solution:
    def getFinalState(self, nums: list[int], k: int, multiplier: int) -> list[int]:

        hq = [(v, i) for i, v in enumerate(nums)]
        heapq.heapify(hq)

        for _ in range(k):
            _, i = heapq.heappop(hq)

            nums[i] *= multiplier
            heapq.heappush(hq, (nums[i], i))

        return nums

File: cpp/3264.final-array-state-after-k-multiplication-operations-i.cpp
#include <algorithm>
#include <vector>
using std::vector;

class Solution {
public:
  vector<int> getFinalState(vector<int> &nums, int k, int multiplier) {
    for (; k; k--) {
      auto minEl = std::min_element(nums.begin(), nums.end());
      *minEl *= multiplier;
    }

    return nums;
  }
};